src/consumer/subscriptionState.js

```diff
@@ -23,4 +25,15 @@ module.exports = class SubscriptionState {
   paused() {
     return Array.from(this.pausedTopics.values())
   }
+
+  /**
+   * @returns {Array<string>} resumed topics
+   */
+  resumed() {
+    return Array.from(this.resumedTopics.values())
+  }
+
+  ackResumed() {
```

Comment:
This feels like it could be error prone, but I don't have a better suggestion. It just feels a bit arbitrary when the resumed topics are no longer considered resumed. I can't even really put into words what makes a topic not resumed anymore.

Reply:
Yeah, resume and pause aren't async operations, so I can't do it there. Then I needed to flag that some topics were just resumed, and this should only happen once. The code is small and straightforward to read; we can revisit this later.